docs/code/VBufferCareFeeding.md

```diff
@@ -142,129 +136,130 @@ using (IRowCursor cursor = dataview.GetRowCursor(col => col == 5))
 to allocate the arrays every single time, completely defeating the purpose of
 buffer reuse.
 
-## Buffer Re-use as a Developer
-
-Nearly all methods in ML.NET that "return" a `VBuffer<T>` do not really return
-a `VBuffer<T>` *at all*, but instead have a parameter `ref VBuffer<T> dst`,
-where they are expected to put the result. See the above example, with the
-`getter`. A `ValueGetter` is defined:
-
-```csharp
-public delegate void ValueGetter<TValue>(ref TValue value);
-```
-
-Let's describe the typical practice of "returning" a `VBuffer` in, say, a
-`ref` parameter named `dst`: if `dst.Indices` and `dst.Values` are
-sufficiently large to contain the result, they are used, and the value is
-calculated, or sometimes copied, into them. If either is insufficiently large,
-then a new array is allocated in its place. After all the calculation happens,
-a *new* `VBuffer` is constructed and assigned to `dst`. (And possibly, if they
-were large enough, using the same `Indices` and `Values` arrays as were passed
-in, albeit with different values.)
-
-`VBuffer`s can be either sparse or dense. However, even when returning a dense
-`VBuffer`, you would not discard the `Indices` array of the passed in buffer,
-assuming there was one. The `Indices` array was merely larger than necessary
-to store *this* result: that you happened to not need it this call does not
-justify throwing it away. We don't care about buffer re-use just for a single
-call, after all! The dense constructor for the `VBuffer` accepts an `Indices`
-array for precisely this reason!
-
-Also note: when you return a `VBuffer` in this fashion, the caller is assumed
-to *own* it at that point. This means they can do whatever they like to it,
-like pass the same variable into some other getter, or modify its values.
+##  Buffer Re-use as a Developer
+
+So we've seen what it looks like from the user's point of view, but some words
+on what a (well implemented) `getter` delegate is doing is also worthwhile, to
+understand what is going on here.
+
+`VBuffer<T>` by itself may appear to be a strictly immutable structure, but
+"buffer reuse" implies it is in some way mutable. This mutability is primarily
+accomplished with the `VBufferEditor`. The `VBufferEditor<T>` actually
+strongly resembles a `VBuffer<T>`, *except* instead of being `ReadOnlySpan`
+for values and indices they are `Span`, so, it is mutable.
+
+First, one creates the `VBufferEditor<T>` structure using one of the
+`VBufferEditor.Create` methods. You pass in a `VBuffer<T>` as a `ref`
+parameter. (For this purpose, it is useful to remember that a
+`default(VBuffer<T>)` is itself a completely valid, though completely empty,
+`VBuffer<T>`.) The editor is past that line considered to "own" the internal
+structure of the `VBuffer<T>` you passed in. (So, correct code should not
+continue to use that `VBuffer<T>` structure from that point onwards.)
+
+Then, one parties on the `Span<T>` for values and, if in a sparse application,
```

Comment:
``one parties on the `Span<T>` for values``

=>

``one mutates the `Span<T>` Values``

?

Reply:
Yes, I independently noticed this a few minutes before you commented on it. 😛 I wound up rephrasing the entire sentence rather than just change a single phrase, maybe see if you like the result when I post the next commit...